src/handwritten/_install_daqmx.py

```diff
@@ -117,49 +152,89 @@ def _multi_access_temp_file(
                 ) from e
 
 
-def _load_data(json_data: str) -> Tuple[Optional[str], Optional[str]]:
+@contextlib.contextmanager
+def _multi_access_temp_folder(*, delete: bool = True) -> Generator[str, None, None]:
+    """
+    Context manager for creating a temporary folder.
+
+    """
+    try:
+        temp_folder = tempfile.TemporaryDirectory()
+        _logger.debug("Created temp folder: %s", temp_folder.name)
+    except Exception as e:
+        _logger.info("Failed to create temporary folder.", exc_info=True)
+        raise click.ClickException(
+            f"Failed to create temporary folder '{temp_folder.name}'.\nDetails: {e}"
+        ) from e
+
+    try:
+        yield temp_folder.name
+    finally:
+        if delete:
+            try:
+                _logger.debug("Deleting temp folder: %s", temp_folder.name)
+                shutil.rmtree(temp_folder.name)
+            except ValueError as e:
+                _logger.info("Failed to delete temporary folder.", exc_info=True)
+                raise click.ClickException(
+                    f"Failed to delete temporary folder '{temp_folder.name}'.\nDetails: {e}"
+                ) from e
+
+
+def _load_data(
+    json_data: str, platform: str
+) -> Tuple[Optional[str], Optional[str], Optional[str], Optional[List[str]]]:
     """
     Load data from JSON string and extract Windows metadata.
 
-    >>> _load_data('{"Windows": [{"Location": "path/to/driver", "Version": "1.0"}]}')
+    >>> _load_data('{"Windows": [{"Location": "path/to/driver", "Version": "1.0", "Release": "2024Q1"}]}')
```

Comment:
Is this doctest going to break every time you update the version?

Reply:
No, it is not expected to fail whenever we update the version.

Reply:
ok, I see, you're passing in the metadata to test with